backend/tracim_backend/lib/collaborative_document_edition/collaboration_document_edition.py

```diff
@@ -0,0 +1,124 @@
+import mimetypes
+import os
+from os.path import isfile
+from os.path import join
+import typing
+
+from defusedxml import ElementTree
+import requests
+from sqlalchemy.orm import Session
+
+from tracim_backend import CFG
+from tracim_backend.exceptions import FileTemplateNotAvailable
+from tracim_backend.exceptions import NotAFileError
+from tracim_backend.exceptions import NotReadableFile
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionConfig,
+)
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionFileType,
+)
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionToken,
+)
+from tracim_backend.lib.collaborative_document_edition.models import FileTemplateList
+from tracim_backend.lib.core.content import ContentApi
+from tracim_backend.lib.utils.utils import is_dir_exist
+from tracim_backend.lib.utils.utils import is_dir_readable
+from tracim_backend.lib.utils.utils import is_file_exist
+from tracim_backend.lib.utils.utils import is_file_readable
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import Content
+
+
+class CollaborativeDocumentEditionApi(object):
+    def __init__(self, current_user: typing.Optional[User], session: Session, config: CFG) -> None:
+        self._session = session
+        self._user = current_user
+        self._config = config
+
+    def discover(self) -> typing.List[CollaborativeDocumentEditionFileType]:
```

Comment:
what is discover? Is this a WOPI api ? If so, please comment somewhere that it is related to WOPI? If not, what is this class, actually?

Reply:
need to explicit this with docstring. That's not at all wopi related.

Reply:
Ok, this is the Api like ContentApi. So I suggest you name something like CollaborativeDocumentEditionLib (would be better to make distinction with http apis, like you suggested a few weeks/months ago). By the way, add a small description to the class

Reply:
done